src/plugins/persistence/couch/CouchObjectProvider.js

```diff
@@ -441,84 +421,58 @@ class CouchObjectProvider {
 
             this.changesFeedSharedWorker.port.postMessage({
                 request: 'changes',
-                body,
                 url
             });
         }
     }
 
-    async fetchChanges(url, body) {
-        const controller = new AbortController();
-        const signal = controller.signal;
+    onEventError(error) {
+        console.error('Error on feed', error);
+        if (Object.keys(this.observers).length > 0) {
+            this.observeObjectChanges();
+        }
+    }
+
+    onEventMessage(event) {
+        const object = JSON.parse(event.data);
+        object.identifier = {
+            namespace: this.namespace,
+            key: object.id
+        };
+        let keyString = this.openmct.objects.makeKeyString(object.identifier);
+        let observersForObject = this.observers[keyString];
+
+        if (observersForObject) {
+            observersForObject.forEach(async (observer) => {
+                const updatedObject = await this.get(object.identifier);
+                observer(updatedObject);
+            });
+        }
+    }
 
-        let error = false;
+    fetchChanges(url) {
+        const controller = new AbortController();
+        let couchEventSource;
+        let sourceListener;
 
         if (this.isObservingObjectChanges()) {
             this.stopObservingObjectChanges();
         }
 
         this.stopObservingObjectChanges = () => {
             controller.abort();
+            couchEventSource.removeEventListener('message', sourceListener);
```

Comment:
As with the corresponding comment in CouchChangesFeed, I don't see a corresponding addEventListener call.

Reply:
It's because `onmessage` is effectively doing that:
https://developer.mozilla.org/en-US/docs/Web/API/EventSource/onmessage
I can switch to being more explicit and use:
`couchEventSource.addEventListener('message', sourceListener);`
instead